docs/source/write_first_app.rst

```diff
@@ -101,889 +97,485 @@ bringing up a new one.
-
-
-Second, the application registers and enrolls an application user
------------------------------------------------------------------
+First, the application creates a gRPC connection to the Gateway peer
+--------------------------------------------------------------------
 
-Now that we have the administrator's credentials in a wallet, the application uses the ``admin``
-user to register and enroll an app user which will be used
-to interact with the blockchain network. The section of the application code is shown below.
+The client application establishes a `gRPC <https://grpc.io/>`__ connection to the Gateway peer that it will use to
+transact with the network. To do this it only requires the endpoint address of the Gateway peer and, if the Gateway is
+configured to use TLS, appropriate TLS certificates to establish a secure connection. There is significant overhead
+associated with establishing gRPC connections, so this connection should be retained by the application and shared by
+all Gateway connections to the Gateway peer.
+
+.. note:: In order to maintain security of any private data used in transactions, the application should connect to a
+          Gateway peer belonging to the same organization as the client identity. If the client identity's organization
+          does not host any peers then a Gateway peer hosted by an upstream organization through which the client's
+          organization is authorized to transact should be used.
+
+The TypeScript application creates a gRPC connection using the TLS certificate of the Certificate Authority for the
+Gateway peer's organization, so that the authenticity of the peer's credentials can be verified. Since the Docker
+container hosting the Gateway peer is reached at a ``localhost`` address, a gRPC option is specified to force this
+endpoint address to be interpreted as the hostname in the peer's TLS certificate, allowing the TLS connection to be
+successfully established.
+
+.. code-block:: TypeScript
+
+  const peerEndpoint = 'localhost:7051';
+
+  async function newGrpcConnection(): Promise<grpc.Client> {
+      const tlsRootCert = await fs.readFile(tlsCertPath);
+      const tlsCredentials = grpc.credentials.createSsl(tlsRootCert);
+      return new grpc.Client(peerEndpoint, tlsCredentials, {
+          'grpc.ssl_target_name_override': 'peer0.org1.example.com',
+      });
+  }
 
-.. code:: bash
+Second, the application creates a Gateway connection
+----------------------------------------------------
 
-  // in a real application this would be done only when a new user was required to be added
-  // and would be part of an administrative flow
-  await registerAndEnrollUser(caClient, wallet, mspOrg1, org1UserId, 'org1.department1');
+The application then creates a Gateway connection, which the application uses to access any of the networks
+(analogous to channels) accessible to the Gateway peer, and smart contracts deployed to those networks. A Gateway
```

Comment:
I think of a channel as more of a "sub-network" in the single overall network. 
Is this connection different than the gRPC connection above?

Reply:
It is different to a gRPC connection. I've reworded to (hopefully) make it clearer.